### ISSUE TITLE
[Backlog]: `collections` crate benchmarking

### ISSUE THREAD

**Issue by berlin-with0ut-return** [OpenDevicePartnership/patina#147]
### Feature Overview

Evaluate and write documentation for the benchmarks in the `collections` crate.

### Urgency

None

### Are you going to implement the feature request?

None

### Do you need maintainer feedback?

None

**Comment by makubacki** [comment]
@Javagedes, is this work still planned?

**Comment by Javagedes** [comment]
@makubacki Yes I'll add it back to my queue. Forgot about it. Thanks for the reminder.
